### PR TITLE
[MM-48028] add intergration section to the preview page

### PR DESCRIPTION
**Pull request by mhd-sln**
#### Summary
Adds integration section for the preview page shown during the work template flow.
(NOTE: In the figma we have category/tag for each plugin but since currently the plugins do not have any tag/category we have no way to show it as such in the UI.)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48028

#### Screenshots

**System user more than one plugin missing**
Screenshot 2022-11-21 at 20 54 28

**System user one plugin missing**
Screenshot 2022-11-21 at 20 55 36

**Admin user **
Screenshot 2022-11-21 at 20 54 47

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/worktemplates/components/preview/section.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
+@import 'utils/variables';
```
Comment (JulienTant) [inline components/worktemplates/components/preview/section.scss:1]: The rest of the feature uses styled-components, it does not really make sense to start using external CSS file in the middle of it 🤔

File: components/worktemplates/components/preview.tsx [hunk 2]
```diff
@@ -97,12 +114,38 @@ const Preview = ({template, className}: PreviewProps) => {
...
+    useEffect(() => {
+        const intg =
+            availableIntegrations?.
+                flatMap((integration) => {
+                    return plugins.reduce((acc: Integration[], curr) => {
+                        if (curr.manifest.id === integration.id) {
+                            acc.push({
+                                ...integration,
+                                name: curr.manifest.name,
+                                description: curr.manifest.description,
```
Comment (JulienTant) [inline components/worktemplates/components/preview.tsx:132]: I don't think we need the description

File: components/worktemplates/components/preview/section.tsx [hunk 3]
```diff
@@ -3,51 +3,138 @@
...
                 {props.message}
             </p>
             <span className='included-title'>{formatMessage({id: 'worktemplates.preview.section.included', defaultMessage: 'Included'})}</span>
-            <ul>
-                {props.items.map((c) => (
-                    <li key={c.id}>
-                        <a
-                            href='#'
-                            onClick={(e) => updateIllustration(e, c.illustration || '')}
+            { viewIntegrations(props) }
```
Comment (JulienTant) [inline components/worktemplates/components/preview/section.tsx:54]: nit: the name of the method leads us the believe we'll only display integration

File: components/worktemplates/components/preview.tsx [hunk 4]
```diff
@@ -58,6 +71,10 @@ const Preview = ({template, className}: PreviewProps) => {
         };
     });
 
+    useEffect(() => {
+        dispatch(fetchListing());
+    }, [dispatch]);
```
Comment (JulienTant) [inline components/worktemplates/components/preview.tsx:76]: I'm not sure if it's a standard or good practice to observe dispatch? In theory, we want to retrieve this only once on the first load.

```suggestion
    }, []);
```

File: components/worktemplates/components/preview/section.tsx [hunk 5]
```diff
@@ -3,51 +3,138 @@
...
+
+        </div>
+    );
+};
+
+const IntegrationsPreview = ({items}: IntegrationPreviewSectionProps) => {
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
```
Comment (JulienTant) [inline components/worktemplates/components/preview/section.tsx:61]: We should look specifically for a role with access to plugin install and not just sysadmins.

**Review by JulienTant — CHANGES_REQUESTED** [review]
Look good overall, just a few things that need change/polish. Good job @mhd-sln

File: components/worktemplates/components/preview.tsx (hunk 4 above)
Comment (mhd-sln) [reply components/worktemplates/components/preview.tsx:76]: I am used to adding dispatch in the dependency list as usually the lint complains about it but for our application it doesn't so I will remove it.

File: components/worktemplates/components/preview.tsx [hunk 6]
```diff
@@ -97,12 +114,37 @@ const Preview = ({template, className}: PreviewProps) => {
...
+                        }
+                        return acc;
+                    }, [] as Integration[]);
+                }).sort((first: Integration) => {
+                    return first.installed ? -1 : 1;
+                });
+        if (intg?.length) {
+            setIntegrations(intg);
+        }
+    }, [plugins]);
```
Comment (julmondragon) [inline components/worktemplates/components/preview.tsx:146]: Shouldn't `availableIntegrations` be added as dependency?

**Review by julmondragon — APPROVED** [review]
LGTM @mhd-sln !

nit: Is it expected that all the modal content scrolls? Maybe only the accordion should?
Nov-22-2022 15-36-49

**Comment by mhd-sln** [comment]
> LGTM @mhd-sln !
> 
> nit: Is it expected that all the modal content scrolls? Maybe only the accordion should? Nov-22-2022 15-36-49 [ Nov-22-2022 15-36-49 ](https://user-images.githubusercontent.com/79058848/203416759-5d64d9ee-3159-4b80-80d3-2261bb4439be.gif) [ ](https://user-images.githubusercontent.com/79058848/203416759-5d64d9ee-3159-4b80-80d3-2261bb4439be.gif)

Yes, according to the design spec all the modal content scrolls.

File: components/worktemplates/components/preview/section.tsx [hunk 7]
```diff
@@ -1,53 +1,235 @@
...
+const IntegrationsPreview = ({items}: IntegrationPreviewSectionProps) => {
+    const systemPermissions = useSelector(getMySystemPermissions);
+    const [pluginInstallationPossible, setPluginInstallationPossible] = useState(false);
+
+    useEffect(() => {
+        for (const systemPermission of [...systemPermissions]) {
+            if (systemPermission.includes('sysconsole_write_plugins')) {
+                setPluginInstallationPossible(true);
+            }
+        }
```
Comment (julmondragon) [inline components/worktemplates/components/preview/section.tsx:69]: nit: What about using `haveISystemPermission` https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/selectors/entities/roles_helpers.ts#L43-L45

**Review by JulienTant — APPROVED** [review]
LGTM! I think it would be beneficial to follow Julian remark to use HaveIPermission but aside from that, it looks good!

**Pull request merged**